### DISCUSSION TITLE
Fixed size watermarks

### DISCUSSION THREAD

**Discussion by suntong** [ImageMagick/ImageMagick#6600] [Help] [answered]
### ImageMagick version

7.1.1-15

### Operating system, version and so on

Debian GNU/Linux 12 (bookworm)

### Description

Continue on with #6594, I want to add a watermark image to the SE corner of another image.

The problem is that my watermark image is fixed size, whereas the target image may vary hugely in size, resulting the added watermarks sometimes small and sometimes huge, or even overflows.

How can I do the calculation on the fly so that the added watermarks is always taking a fixed size/portion of the end result, say 3/4 of its page-width. Somebody gives a single command/script please. thx.

**Comment by snibgo** [comment, marked as answer]
You can resize the watermark, for example so it fits into a box that is 20% of the width and 25% of the height of the main image. Windows BAT script:
```
magick ^
  input.png ^
  -set option:WW %%[fx:w*0.20] ^
  -set option:HH %%[fx:h*0.25] ^
  ( wm.png ^
    -resize %%[WW]x%%[HH] ^
  ) ^
  -gravity SouthEast ^
  -compose Over -composite ^
  out.png
```

**Reply by suntong** [reply]
Thanks,

> Windows BAT script

- The `-set option:` will be working under Linux all the same, or I need to change it to `export` or something?
- The `%%` is a Windows BAT specify symbol or `magick` specify one?

All in all, if I want to use it under Linux, what are the things that I need to change?

**Reply by suntong** [reply]
Hi @snibgo, if I need to move the watermark off the SouthEast corner a bit, say 40+60, how should I change the above command? thx.

**Comment by snibgo** [comment]
I showed a single command, using the syntax needed for Windows BAT scripts. "`-set option:`" is part of the `magick` command.

For bash: change the line-continuation from caret ^ to backslash \\, change each doubled %% to single %, and escape the parentheses \\( and \\).

**Comment by snibgo** [comment]
> if I need to move the watermark off the SouthEast corner a bit, say 40+60 ...

One way is to insert "`-geometry +40+60`" before "`-compose Over -composite`".
